torch/_refs/__init__.py

```diff
@@ -3302,29 +3332,38 @@ def new_ones(
     device: Optional[torch.device] = None,
     pin_memory: bool = False,
 ) -> TensorLikeType:
-    r = a.new_empty(
-        size, dtype=dtype, layout=layout, device=device, pin_memory=pin_memory
+    dtype = a.dtype if dtype is None else dtype
+    layout = a.layout if layout is None else layout
+    device = a.device if device is None else device
+
+    return torch.full(
+        size, 1, dtype=dtype, layout=layout, device=device, pin_memory=pin_memory
     )
-    r.fill_(1)
-    return r
 
 
 @register_decomposition(torch.ops.aten.new_full)
 def new_full(
     a: TensorLikeType,
     size: ShapeType,
-    fill_value: NumberType,
+    fill_value: Union[int, float, bool],
```

Comment:
We should cite an issue for this in a comment. Seems like `new_full()` should allow complex fill values.